js/runtime.go

```diff
@@ -124,21 +124,27 @@ func (r *Runtime) extractOptions(exports otto.Value, opts *lib.Options) error {
 }
 
 func (r *Runtime) loadFile(filename string) (otto.Value, error) {
```

Comment:
That code did not take into account the fact that users may specify the input type with --type. Below is an updated version. As far as I can tell, the error modes we can end up with is if/when:
- A user tries to load a file that has a name that is a valid URL (e.g. the file is called 'http://example.com'), in which case the file will not be opened. Will work if the user specifies --type js, but not if type is auto or 'url'. Unlikely, but may happen.
- A user tries to provide JS code directly on the command line, but there is a file that has the same name as the code provided (extremely unlikely)

```
// if special case name “-“ will always cause src data to be read from file STDIN
// The data read must be URLs or JS code - it cannot be file names
if arg == “-“ {
   src = read_from_stdin()
} else {
   // Deduce and load src data
   switch (type) {
      case TypeAuto:
         if looks_like_url(arg) {  // always try to parse as URL string first
            src = arg
            type = TypeURL
         } else {
            // Otherwise, check if it is a file name and we can load the file
            src, err = read_from_file(arg)
            if err != nil {  // if we fail to open file, we assume the arg is JS code
               src = arg
               type = TypeJS
            }
        }
      case TypeURL:
         // We try to use —type url args as URLs directly first
         if looks_like_url(arg) {
            src = arg
         } else {  // if that didn’t work, we try to load a file with URLs
            src, err = read_from_file(arg)
         }
      case TypeJS:
         // —type js args we try to use as file names first
         src, err = read_from_file(arg)
         if err != nil {  // and if that didn’t work, we assume the arg itself is JS code
            src = arg
         }
   }
}
// Now we should have some src data and in most cases a type also. If we
// don’t have a type it means we read from STDIN or from a file and the user
// specified type == TypeAuto. This means we need to try and auto-detect type
if (type == TypeAuto) {
   if looks_like_url(src) {
      type = TypeURL
   } else {
      type = TypeJS
   }
}
```